controllers/pattern_controller.go

```diff
@@ -208,19 +190,71 @@ func (r *PatternReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ct
 
 	logOnce("namespace found")
 
+	if qualifiedInstance.Spec.GitConfig.UseInClusterFork {
+		// TargetRepo not defined, instantiate a gitea server and fork origin
```

Comment:
Why do we need UseInClusterFork here?
Is that not sensible to imply based on `SourceRepo != nil  && TargetRepo == nil` ?

Reply:
To make it explicit that you want to have your own in cluster replica. Making it explicit ensures that the user is aware that they will need to fetch the repo's URL from a field in the pattern status and the secret from the gitea namespace.